src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORDS_LENGTH = 6;
+    private static final int MIN_USERS_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("Pls create your account. "
+                    + "Enter your Login, Password and Age");
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Pls enter your login");
+        }
+        if (user.getAge() < MIN_USERS_AGE || user.getAge() == null) {
+            throw new RuntimeException("We so sorry, but your age should be than "
```

Comment:
Who are you apologising to? We write these messages for the programmers who will then use our code. Write it in a way that's informative to you.